src/fromager/external_commands.py

```diff
@@ -98,8 +100,31 @@ def run(
             stdin=stdin,
         )
         output = completed.stdout.decode("utf-8") if completed.stdout else ""
+
+    # Get package name from context to add to each line for greppability
+    try:
+        req = log.requirement_ctxvar.get()
+        line_prefix = f"{req.name}: "
+    except LookupError:
+        line_prefix = ""
+
+    # Prefix output lines with visual marker and package name for greppability
+    formatted_output = None
+    if output:
+        output_prefix = line_prefix + "  > "
+        prefixed_output = "  > " + output.replace("\n", f"\n{output_prefix}")
+        formatted_output = f"{line_prefix}{prefixed_output}"
+
     if completed.returncode != 0:
-        logger.error("%s failed with %s", cmd, output)
+        # Build complete error message with command, exit code, and output
+        message = (
+            f"command failed with exit code {completed.returncode}: {shlex.join(cmd)}"
+        )
```

Comment:
Don't use f-strings to format log output. You should always pass a format string and individual arguments to a logger function.

Reply:
This is fixed in the latest version so I am resolving this thread